### PR TITLE
Run api calls from UIThread

### PR DESCRIPTION
**Pull request by pmogollons**
* Simplified calls when possible 
* There are still a couple of methods that might need to run from UIThread but still work as is
* Reordered and grouped methods
* Added listener for onMyAudioSourceTypeChanged

### CODE REVIEW

**Comment by pmogollons** [comment]
Probably we should wrap this in the UIThread, not sure what zoom thinks is an API call.

```javascript
    ZoomSDK zoomSDK = ZoomSDK.getInstance();

    if (!zoomSDK.isInitialized()) {
      return;
    }
```

File: android/src/main/java/ch/milosz/reactnative/RNZoomUsModule.java [hunk 1]
```diff
@@ -392,9 +390,12 @@ public void isMeetingConnected(Promise promise) {
...
 
-    final InMeetingService inMeetingService = zoomSDK.getInMeetingService();
-
-    promise.resolve(inMeetingService.isMeetingConnected());
+    UiThreadUtil.runOnUiThread(new Runnable() {
+      @Override
+      public void run() {
+        promise.resolve(zoomSDK.getInMeetingService().isMeetingConnected());
+      }
+    });
```
Comment (mieszko4) [inline android/src/main/java/ch/milosz/reactnative/RNZoomUsModule.java:398]: this and `isMeetingHost` actually does not need to be wrapped - I've tested it myself.
But I guess that's fine to have this pattern consistently across this file

File: android/src/main/java/ch/milosz/reactnative/RNZoomUsModule.java [hunk 2]
```diff
@@ -781,143 +820,155 @@ private void unregisterListener() {
...
-    sendEvent("MeetingEvent", "userJoin", list);
+    sendEvent("MeetingEvent", "userJoin", userList);
   }
 
   @Override
-  public void onMeetingUserLeave(List<Long> list) {
+  public void onMeetingUserLeave(List<Long> userList) {
     updateVideoView();
-    sendEvent("MeetingEvent", "userLeave", list);
+    sendEvent("MeetingEvent", "userLeave", userList);
```
Comment (mieszko4) [inline android/src/main/java/ch/milosz/reactnative/RNZoomUsModule.java:844]: I suggest: `userList` -> `userIdList`

**Review by mieszko4 — APPROVED** [review]
Great, I think we can merge this!
Just please in the future do try to split this code movement by more commits because it was very hard to track the changes for me.

**Comment by mieszko4** [comment]
@pmogollons I see this is a draft. Did you have anything else in mind for doing in this pr?

**Comment by pmogollons** [comment]
Yes, I wanted to check with you if we should wrap all "api calls" in UIThread even if they are still working as is, but you already said yes for consistency so Ill make the remaining changes today.

**Comment by mieszko4** [comment]
> Yes, I wanted to check with you if we should wrap all "api calls" in UIThread even if they are still working as is, but you already said yes for consistency so Ill make the remaining changes today.

Awesome, thnx! Before you do that please merge/rebase the current master

**Comment by pmogollons** [comment]
I finally didnt went with the other changes, those are working correctly as is so for now I dont feel we should change it. I removed the comments from those methods, you can check which ones in the last commit.

Ill finish testing and come back here to comment if everything is working.

**Comment by pmogollons** [comment]
I just found out that share screen is not implemented correctly, but it has nothing to do with UIThread, so ill create an issue and a PR for that. Appart from that its looking good.

**Pull request merged**